src/main/java/de/plushnikov/intellij/plugin/extension/LombokHighlightErrorFilter.java

```diff
@@ -51,30 +61,96 @@ public boolean accept(@NotNull HighlightInfo highlightInfo, @Nullable PsiFile fi
       return true;
     }
 
-    String description = StringUtil.notNullize(highlightInfo.getDescription());
-    if (HighlightSeverity.ERROR.equals(highlightInfo.getSeverity())) {
+    // check exceptions for highlights
+    boolean acceptHighlight = registeredFilters
+      .getOrDefault(highlightInfo.getSeverity(), Collections.emptyMap())
+      .getOrDefault(highlightInfo.type.getAttributesKey(), Collections.emptyList())
+      .stream()
+      .filter(filter -> filter.descriptionCheck(highlightInfo.getDescription()))
+      .allMatch(filter -> filter.accept(highlightedElement));
+
+    // check if highlight was filtered
+    if (!acceptHighlight) {
+      return false;
+    }
 
-      // Handling LazyGetter
-      if (uninitializedField(description) && LazyGetterHandler.isLazyGetterHandled(highlightedElement)) {
+    // handle rest cases
+    String description = highlightInfo.getDescription();
+    if (HighlightSeverity.ERROR.equals(highlightInfo.getSeverity())) {
+      //Handling onX parameters
+      if (OnXAnnotationHandler.isOnXParameterAnnotation(highlightInfo, file)
+        || OnXAnnotationHandler.isOnXParameterValue(highlightInfo, file)
+        || (description != null && LOMBOK_ANY_ANNOTATION_REQUIRED.matcher(description).matches())) {
         return false;
       }
+    }
 
-      //Handling onX parameters
-      return !OnXAnnotationHandler.isOnXParameterAnnotation(highlightInfo, file)
-        && !OnXAnnotationHandler.isOnXParameterValue(highlightInfo, file)
-        && !LOMBOK_ANY_ANNOTATION_REQUIRED.matcher(description).matches();
+    // register different quick fix for highlight
+    registeredHooks
+      .getOrDefault(highlightInfo.getSeverity(), Collections.emptyMap())
+      .getOrDefault(highlightInfo.type.getAttributesKey(), Collections.emptyList())
+      .stream()
+      .filter(filter -> filter.descriptionCheck(highlightInfo.getDescription()))
+      .forEach(filter -> filter.processHook(highlightedElement, highlightInfo));
+
+    return true;
+  }
+
+  private enum LombokHighlightFixHook {
```

Comment:
I'm unsure, that this is good approach... It very surprising for me to have a Filter-Class, that effectively changed something on "HighlightsInfo"-Objects inside.

Reply:
Best approach could be to ask JetBrains to add new ExtensionPoint to com.intellij.codeInsight.daemon.impl.analysis.HighlightFixUtil#registerUnhandledExceptionFixes, to be able to register custom QuickFix.

Or to implement a new inspection, that checks for unhandled exception and register quickfix there. But this would double a lot of logic for/from HighlightUtil.checkUnhandledExceptions calls...

Reply:
> I'm unsure, that this is good approach... It very surprising for me to have a Filter-Class, that effectively changed something on "HighlightsInfo"-Objects inside.

It's the same highlightInfo object. When somebody decides to register a new highlight it goes throw filter, that's why we can add anything we want in this object. I dunno it's a good idea or not. It should be discussed with Jetbrains. But if we perceive this extension point as the ability to modify and drop all registered highlights, it will be the right point for modifications.

Does additional inspection decrease performance?

Reply:
After thinking for a long time, I decided to merge it. :-)